connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+    """Google Drive"""
+
+    name = "Google Drive"
+    service_type = "google_drive"
+
+    def __init__(self, configuration):
+        """Set up the connection to the Google Drive Client.
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Google Drive.
+
+        Returns:
+            dict: Default configuration.
+        """
+        default_credentials = {
+            "type": "service_account",
+            "project_id": "project_id",
+            "private_key_id": "abc",
+            "private_key": open(
+                os.path.abspath(DEFAULT_PEM_FILE)
+            ).read(),  # TODO: change this and provide meaningful defaults
+            "client_email": "123-abc@developer.gserviceaccount.com",
+            "client_id": "123-abc.apps.googleusercontent.com",
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "http://localhost:10339/token",
+        }
+        return {
+            "service_account_credentials": {
+                "display": "textarea",
+                "label": "Google Drive service account JSON",
+                "order": 1,
+                "type": "str",
+                "value": json.dumps(default_credentials),
+            },
+            "retry_count": {
+                "default_value": DEFAULT_RETRY_COUNT,
+                "display": "numeric",
+                "label": "Maximum retries for failed requests",
+                "order": 2,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": DEFAULT_RETRY_COUNT,
+            },
+        }
+
+    @cached_property
+    def _google_drive_client(self):
+        """Initialize and return the GoogleDriveClient
+
+        Returns:
+            GoogleDriveClient: An instance of the GoogleDriveClient.
+        """
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = get_pem_format(
+                key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        return GoogleDriveClient(
+            json_credentials=json_credentials,
+            retry_count=self.configuration["retry_count"],
+        )
+
+    async def ping(self):
+        """""Verify the connection with Google Drive""" ""
+        if RUNNING_FTEST:
+            return
+
+        try:
+            await anext(
+                self._google_drive_client.api_call(
+                    resource="about", method="get", fields="kind"
+                )
+            )
```

Comment:
It was needed because `api_call()` was returning async generator for both paginated and non-paginated calls, await anext was get the first (and single) object from the async iterator. 

Following suggestions from other comments I'm splitting the api_call function so that the paginated one will be returning async generator whereas the non-paginated func will be returning awaitable result.